### PR TITLE
Update to pulpcore 3.17

### PR DESCRIPTION
**Pull request by bmclaughlin**
~~Needs https://github.com/pulp/pulpcore/pull/1780 when released~~

Issue: AAH-1090

### CODE REVIEW

**Review by rochacbruno — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -89,11 +89,11 @@ def run(self):
 
 requirements = [
     "galaxy-importer==0.4.2",
-    "pulpcore>=3.15.0,<3.16.0",
-    "pulp-ansible>=0.10.1,<0.11.0",
+    "pulpcore>=3.17.1,<3.18.0",
+    "pulp-ansible>=0.11.0,<0.12.0",
```
Comment (awcrosby) [inline setup.py:93]: should this minimum `pulp-ansible>=0.11.0` and template_config.yml value `branch: 0.11.1` match?

Comment (bmclaughlin) [reply setup.py:93]: good catch!

**Review by awcrosby — APPROVED** [review]

**Comment by fao89** [comment]
We are currently having this issue on pulp_installer CI:
```
Traceback (most recent call last):
    File "/usr/local/lib/pulp/lib64/python3.8/site-packages/django/db/models/query.py", line 581, in get_or_create
      return self.get(**kwargs), False
    File "/usr/local/lib/pulp/lib64/python3.8/site-packages/django/db/models/query.py", line 435, in get
      raise self.model.DoesNotExist(
  __fake__.DoesNotExist: AccessPolicy matching query does not exist.
  
  During handling of the above exception, another exception occurred:
  
  Traceback (most recent call last):
    File "/usr/local/lib/pulp/bin/pulpcore-manager", line 33, in <module>
      sys.exit(load_entry_point('pulpcore', 'console_scripts', 'pulpcore-manager')())
    File "/var/lib/pulp/devel/pulpcore/pulpcore/app/manage.py", line 11, in manage
      execute_from_command_line(sys.argv)
    File "/usr/local/lib/pulp/lib64/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
      utility.execute()
    File "/usr/local/lib/pulp/lib64/python3.8/site-packages/django/core/management/__init__.py", line 413, in execute
      self.fetch_command(subcommand).run_from_argv(self.argv)
    File "/usr/local/lib/pulp/lib64/python3.8/site-packages/django/core/management/base.py", line 354, in run_from_argv
      self.execute(*args, **cmd_options)
    File "/usr/local/lib/pulp/lib64/python3.8/site-packages/django/core/management/base.py", line 398, in execute
      output = self.handle(*args, **options)
    File "/usr/local/lib/pulp/lib64/python3.8/site-packages/django/core/management/base.py", line 89, in wrapped
      res = handle_func(*args, **kwargs)
    File "/usr/local/lib/pulp/lib64/python3.8/site-packages/django/core/management/commands/migrate.py", line 268, in handle
      emit_post_migrate_signal(
    File "/usr/local/lib/pulp/lib64/python3.8/site-packages/django/core/management/sql.py", line 42, in emit_post_migrate_signal
      models.signals.post_migrate.send(
    File "/usr/local/lib/pulp/lib64/python3.8/site-packages/django/dispatch/dispatcher.py", line 180, in send
      return [
    File "/usr/local/lib/pulp/lib64/python3.8/site-packages/django/dispatch/dispatcher.py", line 181, in <listcomp>
      (receiver, receiver(signal=self, sender=sender, **named))
    File "/var/lib/pulp/devel/galaxy_ng/galaxy_ng/app/signals/handlers.py", line 59, in set_pulp_container_access_policies
      policy, created = AccessPolicy.objects.update_or_create(
    File "/usr/local/lib/pulp/lib64/python3.8/site-packages/django/db/models/manager.py", line 85, in manager_method
      return getattr(self.get_queryset(), name)(*args, **kwargs)
    File "/usr/local/lib/pulp/lib64/python3.8/site-packages/django/db/models/query.py", line 608, in update_or_create
      obj, created = self.select_for_update().get_or_create(defaults, **kwargs)
    File "/usr/local/lib/pulp/lib64/python3.8/site-packages/django/db/models/query.py", line 583, in get_or_create
      params = self._extract_model_params(defaults, **kwargs)
    File "/usr/local/lib/pulp/lib64/python3.8/site-packages/django/db/models/query.py", line 634, in _extract_model_params
      raise exceptions.FieldError(
  django.core.exceptions.FieldError: Invalid field name(s) for model AccessPolicy: 'permissions_assignment'.
  ```
  https://github.com/pulp/pulp_installer/runs/4704427904?check_suite_focus=true#step:7:2573
  
  I believe this PR fixes this issue, I'm commenting here to get notified when this PR gets merged, so I can check if it fixes the pulp_installer CI issue

**Comment by fao89** [comment]
Unfortunately, it didn't fix the CI issue:
https://github.com/pulp/pulp_installer/runs/4705652184?check_suite_focus=true#step:7:1764
I'm wondering if it is related to the new roles on pulp_container
FYI @mdellweg

**Comment by mdellweg** [comment]
I think the problem arises from: galaxy_ng/app/signals/handlers.py", line 59, in set_pulp_container_access_policies
In https://github.com/ansible/galaxy_ng/blob/master/galaxy_ng/app/access_control/statements/pulp_container.py the fields `permissions_assignment` should be renamed to `creation_hooks`. I wonder why the ci didn't find it here. Does in not install the pulp_container plugin and run migrations?

Also i noticed, that the molecule file may refer to improper `host/group_vars`:
https://github.com/pulp/pulp_installer/blob/main/molecule/source-static/molecule.yml#L55
I was wondering why galaxy was installed at all when only file was requested. XD

**Comment by fao89** [comment]
> I think the problem arises from: galaxy_ng/app/signals/handlers.py", line 59, in set_pulp_container_access_policies In https://github.com/ansible/galaxy_ng/blob/master/galaxy_ng/app/access_control/statements/pulp_container.py the fields `permissions_assignment` should be renamed to `creation_hooks`. I wonder why the ci didn't find it here. Does in not install the pulp_container plugin and run migrations?
> 
> Also i noticed, that the molecule file may refer to improper `host/group_vars`: https://github.com/pulp/pulp_installer/blob/main/molecule/source-static/molecule.yml#L55 I was wondering why galaxy was installed at all when only file was requested. XD

I'll investigate both CI's to get a consistent behavior

**Pull request merged**